python/cudf/cudf/pylibcudf_tests/test_string_find.py

```diff
@@ -0,0 +1,118 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+import numpy as np
+import pyarrow as pa
+import pytest
+from utils import assert_column_eq
+
+import cudf._lib.pylibcudf as plc
+from cudf._lib.scalar import DeviceScalar
+
+
+@pytest.fixture(scope="module")
+def string_col():
+    return pa.array(
+        ["AbC", "de", "FGHI", "j", "kLm", "nOPq", None, "RsT", None, "uVw"]
+    )
+
+
+@pytest.mark.parametrize("target", ["a", ""])
+def test_find(string_col, target):
+    plc_col = plc.interop.from_arrow(string_col)
+    got = plc.strings.find.find(
+        plc_col, DeviceScalar(target, dtype=np.dtype("object")).c_value, 0, -1
+    )
+    expected = pa.Array.from_pandas(
+        string_col.to_pandas().str.find(target), type=pa.int32()
+    )
+    assert_column_eq(got, expected)
+
+
+def test_find_column(string_col):
+    plc_col = plc.interop.from_arrow(string_col)
+    target_col = plc.interop.from_arrow(
+        pa.array(["A", "d", "F", "j", "k", "n", None, "R", None, "u"])
+    )
+    expected = pa.array([0, 0, 0, 0, 0, 0, None, 0, None, 0], type=pa.int32())
+    got = plc.strings.find.find(plc_col, target_col, 0)
+    assert_column_eq(got, expected)
+
+
+@pytest.mark.parametrize("target", ["a", ""])
+def test_rfind(string_col, target):
+    plc_col = plc.interop.from_arrow(string_col)
+    got = plc.strings.find.rfind(
+        plc_col, DeviceScalar(target, dtype=np.dtype("object")).c_value, 0, -1
+    )
+    expected = pa.Array.from_pandas(
+        string_col.to_pandas().str.rfind(target), type=pa.int32()
+    )
```

Comment:
I'd love to avoid using pandas. I'd rather hardcode an expected output than use pandas for this testing, I think? Another option could be doing something clever like reversing the string (pyarrow does support that) and then doing a `find_substring`. I'm not sure, WDYT?

Reply:
Dropped to python stringops in https://github.com/rapidsai/cudf/pull/15604/commits/6cd87476824f087b70740229a80da0dcae41cb2c